File: 2020/03 - Mar/122 Stock_2.java
class Solution {
    public int maxProfit(int[] prices) {

        """
        Runtime: 1 ms, faster than 91.80% of Java online submissions for Best Time to Buy and Sell Stock II.
        Memory Usage: 42.5 MB, less than 5.71% of Java online submissions for Best Time to Buy and Sell Stock II.
        """
        if (prices == null || prices.length <= 1) {
            return 0;
        }
        
        int result = 0;
        int buy_price = prices[0];
        for (int i = 1; i < prices.length; i++) {
            int new_price = prices[i];
            if (new_price > buy_price) { result += new_price - buy_price; }
            buy_price = new_price;
        }
        return result;
    }
}

File: 2020/09 - September/122 Stock_2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        """Max profit of +inf buy/sell(s)"""
        
        """
        Runtime: 60 ms, faster than 85.85% of Python3 online submissions for Best Time to Buy and Sell Stock II.
        Memory Usage: 15.1 MB, less than 57.64% of Python3 online submissions for Best Time to Buy and Sell Stock II.
        """
        curProf = 0
        curBuy = float('inf')
        for p in prices:
            if p > curBuy:
                curProf += (p - curBuy)
            curBuy = p

        return curProf